pkg/controller/cluster/cluster_reconcile.go

```diff
@@ -141,3 +161,24 @@ func (r *ReconcileCluster) updateTenantGitRepo(tenant types.NamespacedName, file
 
 	return nil
 }
+
+func (r *ReconcileCluster) getServiceAccountToken(instance *synv1alpha1.Cluster) (string, error) {
+	secrets := &corev1.SecretList{}
+
+	err := r.client.List(context.TODO(), secrets)
+	if err != nil {
+		return "", err
+	}
+
+	sortSecrets := helpers.SecretSortList(*secrets)
+
+	sort.Sort(sort.Reverse(sortSecrets))
+
+	for _, secret := range sortSecrets.Items {
+		if secret.Annotations[corev1.ServiceAccountNameKey] == instance.Name {
```

Comment:
Let's also check for `secret.Type == corev1.SecretTypeServiceAccountToken` and if the secret really has a key `secret.Data["token"]` (if a new secret is created, it doesn't have the token yet)...

Reply:
This returns now a specific error that gets checked in the reconcile function and then ignored. So we don't trigger unnecessary reconciles while waiting for the token to be populated.